MiniApp/Classes/core/Assets/console.js

```diff
@@ -0,0 +1,22 @@
+function log(emoji, type, args) {
+  window.webkit.messageHandlers.logging.postMessage(
+    `${emoji} console.${type}: ${Object.values(args)
+      .map(v => typeof(v) === "undefined" ? "undefined" : typeof(v) === "object" ? JSON.stringify(v) : v.toString())
+      .map(v => v.substring(0, 3000)) // Limit msg to 3000 chars
+      .join(", ")}`
+  )
+}
+
+let originalLog = console.log
+let originalWarn = console.warn
+let originalError = console.error
+let originalDebug = console.debug
+
+console.log = function() { log("📗", "log", arguments); originalLog.apply(null, arguments) }
+console.warn = function() { log("📙", "warning", arguments); originalWarn.apply(null, arguments) }
+console.error = function() { log("📕", "error", arguments); originalError.apply(null, arguments) }
+console.debug = function() { log("📘", "debug", arguments); originalDebug.apply(null, arguments) }
+
+window.addEventListener("error", function(e) {
+   log("💥", "Uncaught", [`${e.message} at ${e.filename}:${e.lineno}:${e.colno}`])
+})
```

Comment:
I think this listener should be `miniapperror` - `error` is too generic so this could conflict with other listeners in used by the mini app.

Also, just to make sure - this will only be available when the Host App is built as debug, right? The end developers of Mini Apps won't see these logs?

Reply:
yep @corycaywood , this script will only be included when the pod (thus the host app) is compiled in debug mode